pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/SnapshotSegmentAbortedTxnProcessorImpl.java

```diff
@@ -384,8 +385,12 @@ public CompletableFuture<Void> clearAbortedTxnSnapshot() {
     }
 
     @Override
-    public long getLastSnapshotTimestamps() {
-        return this.lastSnapshotTimestamps;
+    public void getSnapshotStats(TransactionBufferStats stats) {
+        stats.lastSnapshotTimestamps = this.lastSnapshotTimestamps;
+        TransactionBufferStats.SnapshotStats snapshotStats = new TransactionBufferStats.SnapshotStats();
+        snapshotStats.segmentsSize = this.indexes.size();
+        snapshotStats.unsealedAbortTxnIDs = this.unsealedTxnIds.size();
+        stats.snapshotStats = snapshotStats;
```

Comment:
If there has an in flight task which is typed `write segment`, there is an issue: 
- the variable `unsealedAbortTxnIDs` will be zero
- the variable `segmentsSize` is one less than the correct value

Reply:
In my view, this particular call is only retrieving data at the lastSnapshotTimestamps timestamp. If the user has an expected result, they can make multiple admin calls for that purpose.

Reply:
> In my view, this particular call is only retrieving data at the lastSnapshotTimestamps timestamp. If the user has an expected result, they can make multiple admin calls for that purpose.

If users knew the correct value, they do not need this API anymore, right?

Reply:
Maybe they can make some judgments by the `lastSnapshotTimestamps` timestamp.

Reply:
I believe the most crucial consideration is whether we need to provide full concurrency safety guarantees for this admin API. If it is necessary to ensure that the data obtained by the admin is not affected by concurrent operations, then we must introduce locks for the `generateSnapshotStats` and `putAbortedTxnAndPosition` methods. We also need to consider whether we should introduce locks for the hot path of message sending in this admin API.

Reply:
I can replace ` snapshotStats.segmentsSize = this.indexes.size();` with `snapshotStats.unsealedAbortTxnIDs = this.unsealedTxnIds.size();` which update before persistenting segment, but the concurrent problem still exists.